specs/validator/0_beacon-chain-validator.md

```diff
@@ -0,0 +1,383 @@
+
+_Note:_ This can be looked up in the state using `get_block_root(state, justified_slot)`.
+
+#### Construct attestation
+
+Next the validator creates `attestation`, an [`Attestation`](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#attestation) object.
+
+##### Data
+
+Set `attestation.data = attestation_data` where `attestation_data` is the `AttestationData` object defined in the previous section, [attestation data](#attestation-data).
+
+##### Participation bitfield
+
+* Let `participation_bitfield` be a byte array filled with zeros of length `(len(committee) + 7) // 8`.
+* Let `index_into_committee` be the index into the validator's `committee` at which `validator_index` is located.
+* Set `participation_bitfield[index_into_committee // 8] |= 2 ** (index_into_committee % 8)`.
+* Set `attestation.participation_bitfield = participation_bitfield`.
+
+_Note_: Calling `get_attestation_participants(state, attestation.data, attestation.participation_bitfield)` should return `[validator_index]`.
+
+##### Custody bitfield
+
+* Let `custody_bitfield` be a byte array filled with zeros of length `(len(committee) + 7) // 8`.
+* Set `attestation.custody_bitfield = custody_bitfield`.
+
+_Note:_ This is a stub for phase 0.
+
+##### Aggregate signature
+
+Set `attestation.aggregate_signature = signed_attestation_data` where `signed_attestation_data` is defined as:
+
+```python
+attestation_data_and_custody_bit = AttestationDataAndCustodyBit(
+    attestation.data,
+    False,
+)
+attestation_message_to_sign = hash_tree_root(attestation_data_and_custody_bit)
+
+signed_attestation_data = bls_sign(
+    privkey=validator.privkey,  # privkey store locally, not in state
+    message=attestation_message_to_sign,
+    domain=get_domain(
+        state.fork_data,  # `state` is the state at `head`
+        state.slot,
+        DOMAIN_ATTESTATION,
+    )
+)
+```
+
+## How to avoid slashing
+
+"Slashing" is the burning of some amount of validator funds and immediate ejection from the active validator set. In Phase 0, there are two ways in which funds can be slashed -- [proposal slashing](#proposal-slashing) and [attestation slashing](#casper-slashing). Although being slashed has serious repercussions, it is simple enough to avoid being slashed all together by remaining _consistent_ with respect to the messages you have previously signed.
+
+_Note_: Signed data must be within the same `Fork` context to conflict. Messages cannot be slashed across forks.
+
+### Proposal slashing
+
+To avoid "proposal slashings", a validator must not sign two conflicting [`ProposalSignedData`](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#proposalsigneddata) (suggest renaming `ProposalData`) where conflicting is defined as having the same `slot` and `shard` but a different `block_root`.
```

Comment:
The `shard` in this case is always `BEACON_CHAIN_SHARD_NUMBER`; probably worth clarifying that.

Reply:
I feel like this section is too verbose; the impression you should be giving readers is something like "don't propose two different things in the same slot and you'll be fine, dawg"; making it much longer than that risks diluting the message. I'd actually consider just merging this section in with the Casper slashing section, as you gain succinctness by describing the "save to disk then broadcast" pattern once.